File: src/lib/api/keychain.ts
import { invoke } from "@tauri-apps/api/core";
import type { AppId } from "./types";

export const keychainApi = {
  /**
   * 将 API Key 安全存储到系统 Keychain
   */
  async setApiKey(
    providerId: string,
    appType: AppId,
    apiKey: string,
  ): Promise<void> {
    await invoke("set_api_key", {
      providerId,
      appType,
      apiKey,
    });
  },

  /**
   * 从系统 Keychain 读取 API Key
   */
  async getApiKey(providerId: string, appType: AppId): Promise<string | null> {
    const result = await invoke<string | null>("get_api_key", {
      providerId,
      appType,
    });
    return result;
  },

  /**
   * 从系统 Keychain 删除 API Key
   */
  async deleteApiKey(providerId: string, appType: AppId): Promise<void> {
    await invoke("delete_api_key", {
      providerId,
      appType,
    });
  },
};

File: src-tauri/src/commands/keychain.rs
use crate::error::AppError;
use crate::store::{self, AppState};
use tauri::State;

/// 将 API Key 安全存储到系统 Keychain
#[tauri::command]
pub fn set_api_key(provider_id: String, app_type: String, api_key: String) -> Result<(), String> {
    store::set_api_key(&provider_id, &app_type, &api_key)
}

/// 从系统 Keychain 读取 API Key
#[tauri::command]
pub fn get_api_key(provider_id: String, app_type: String) -> Result<Option<String>, String> {
    store::get_api_key(&provider_id, &app_type)
}

/// 从系统 Keychain 删除 API Key
#[tauri::command]
pub fn delete_api_key(provider_id: String, app_type: String) -> Result<(), String> {
    store::delete_api_key(&provider_id, &app_type)
}
